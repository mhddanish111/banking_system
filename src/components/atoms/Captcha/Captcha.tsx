import React from 'react';
import { isEmpty } from 'lodash';
import ReCAPTCHA from 'react-google-recaptcha';

type captchaProps = {
  setReCaptchaStatus: any;
};

const Captcha: React.FC<captchaProps> = ({ setReCaptchaStatus }) => {
  const siteKey = '6Lc7_dsjAAAAAPAvV1UPgFBBQ3VSoGxZcx3y-uDg';
  const onChange = (value: any): void => {
    let valid = false;
    let showErrorMsg = true;
    if (!isEmpty(value)) {
      valid = true;
      showErrorMsg = false;
    }
    setReCaptchaStatus({ valid, showErrorMsg });
  };
  return <ReCAPTCHA onChange={onChange} sitekey={siteKey} />;
};

export default Captcha;
