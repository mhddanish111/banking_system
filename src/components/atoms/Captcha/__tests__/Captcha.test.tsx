import React from 'react';
import { shallow } from 'enzyme';
import ReCAPTCHA from 'react-google-recaptcha';
import Captcha from '../index';

const props = {
  isoCode: '',
  setReCaptchaStatus: jest.fn(),
};

describe('<Captcha />', () => {
  let CaptchaComponent = '';

  beforeEach(() => {
    CaptchaComponent = shallow(<Captcha {...props} />);
  });

  test('should render correctly', () => {
    expect(CaptchaComponent).toMatchSnapshot();
  });

  test('should call onChange event', () => {
    const setReCaptchaStatus = jest.fn();
    CaptchaComponent.setProps({ setReCaptchaStatus });
    const event = { preventDefault: jest.fn(), value: 'abc' };
    CaptchaComponent.find(ReCAPTCHA).simulate('change', event);
    expect(setReCaptchaStatus).toHaveBeenCalled();
  });
});
