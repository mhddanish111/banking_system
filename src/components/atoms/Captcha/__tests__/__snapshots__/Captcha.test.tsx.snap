// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Captcha /> should render correctly 1`] = `
<AsyncScriptLoader(ReCAPTCHA)
  onChange={[Function]}
  sitekey="6Lc7_dsjAAAAAPAvV1UPgFBBQ3VSoGxZcx3y-uDg"
/>
`;
